### DISCUSSION TITLE
Feature Request: Support duckdb.register on DuckDBPyRelation Class

### DISCUSSION THREAD

**Discussion by swang1001** [duckdb/duckdb#6461] [Ideas]
The new feature released in 0.7.0 that allows incremental query building is very nice: https://duckdb.org/2023/02/13/announcing-duckdb-070.html#python-api-improvements.  However, this requires all required DuckDBPyRelation objects to be in scope. It will be great if `duckdb.register` can be updated to support these objects in addition to Pandas dataframes. More specifically, I am referring to this function: https://github.com/duckdb/duckdb/blob/v0.7.0/tools/pythonpkg/src/pyconnection.cpp#L412

**Comment by Tishj** [comment]
PR is up in #6483
